webapp/packages/plugin-sql-editor/src/SqlEditor/useTools.tsx

```diff
@@ -13,89 +13,78 @@ import { ConnectionInfoResource } from '@cloudbeaver/core-connections';
 
-import type { ISqlEditorTabState } from '../ISqlEditorTabState';
 import { SqlEditorSettingsService } from '../SqlEditorSettingsService';
 import { ScriptImportDialog } from './ScriptImportDialog';
-import type { SqlEditorController } from './SqlEditorController';
 
 interface State {
-  uploadScript: (files: FileList | null) => Promise<void>;
-  downloadScript: () => void;
+  tryReadScript: (file: File, prevScript: string) => Promise<string | null>;
+  readScript: (file: File) => Promise<string | null>;
+  downloadScript: (script: string) => void;
+  isFileValid: (file: File) => boolean;
 }
 
-export function useTools(controller: SqlEditorController, state: ISqlEditorTabState): Readonly<State> {
+export function useTools(connectionId: string | undefined): Readonly<State> {
   const commonDialogService = useService(CommonDialogService);
   const notificationService = useService(NotificationService);
   const connectionInfoResource = useService(ConnectionInfoResource);
   const sqlEditorSettingsService = useService(SqlEditorSettingsService);
 
   return useObservableRef(() => ({
-    async uploadScript(fileList: FileList | null) {
-      if (!fileList) {
-        throw new Error('No files found');
-      }
+    async tryReadScript(file: File, prevScript: string) {
+      const valid = this.isFileValid(file);
 
-      if (!fileList.length) {
-        return;
+      if (!valid) {
+        return null;
       }
 
-      if (this.controller.value.trim()) {
+      if (prevScript) {
         const result = await this.commonDialogService.open(ScriptImportDialog, null);
 
         if (result === DialogueStateResult.Rejected) {
-          return;
+          return null;
         }
 
         if (result !== DialogueStateResult.Resolved && result) {
-          this.downloadScript();
+          this.downloadScript(prevScript);
         }
       }
 
-      const maxSize = this.sqlEditorSettingsService.settings.getValue('maxFileSize');
-      const aboveMaxSize: File[] = [];
-
-      const files = Array.from(fileList).filter(file => {
-        const size = Math.round(file.size / 1000); // kilobyte
-        const above = size > maxSize;
+      return this.readScript(file);
+    },
 
-        if (above) {
-          aboveMaxSize.push(file);
-        }
+    async readScript(file: File) {
+      let script = null;
+      try {
+        const process = getTextFileReadingProcess(file);
+        script = await process.promise;
+      } catch (exception) {
+        this.notificationService.logException(exception, 'Uploading script error');
+      }
 
-        return !above;
-      });
+      return script;
+    },
 
-      if (aboveMaxSize.length > 0) {
-        const nameList = aboveMaxSize.map(file => `${file.name} (${Math.round(file.size / 1000)}KB)`);
-        const message = `Max size: ${maxSize}KB\nFollowing files exceed max size: "${nameList.join(', ')}"`;
+    isFileValid(file: File) {
```

Comment:
better to rename to `checkFileValid` or move notification display logic outside this function